### PR TITLE
fix(combobox): combobox components issues

### PR DESCRIPTION
**Pull request by iropolo**
# Combobox components issues

## :hammer_and_wrench: Type Of Change

- [x] Fix

## :book: Description

Bug fixs [DT-860](https://dialpad.atlassian.net/browse/DT-860)

## :camera: Screenshots / GIFs

image

### CODE REVIEW

File: components/chip/chip.vue [hunk 1]
```diff
@@ -145,6 +145,14 @@ export default {
...
       default: '',
     },
+
+    /**
+     * Additional class name for the span element.
+     */
+    parentClass: {
+      type: [String, Array, Object],
+      default: '',
+    },
```
Comment (braddialpad) [inline components/chip/chip.vue]: Didn't we decide to go with labelClass in this PR? https://github.com/dialpad/dialtone-vue/pull/642 should be the same here.

Comment (iropolo) [reply components/chip/chip.vue]: It is not the same ticket.
Once the previous fix is approved, I will merge it here and it will be updated as we wish.

**Comment by josedialpad** [comment]
The [Lint PR title action](https://github.com/dialpad/dialtone-vue/actions/runs/3742501735/jobs/6353521153) is failing because it's not following all the rules in the [commit convention](https://github.com/dialpad/dialtone-vue/blob/staging/.github/COMMIT_CONVENTION.md) due to capitalizing the subject part of the title. It should be: `fix(combobox): combobox components issues`.

Also since the scope is already defining the component affected by these changes, I'd suggest using something like: `fix(combobox): fix issues` and maybe detail the issues solved if it's not very extensive.

Once you fix the PR title here, please apply the same changes to the [Vue3 PR](https://github.com/dialpad/dialtone-vue/pull/657).

File: recipes/comboboxes/combobox_with_popover/combobox_with_popover.vue [hunk 2]
```diff
@@ -94,6 +94,7 @@
           <div
             v-if="$slots.footer"
             ref="footer"
+            tabindex="0"
```
Comment (braddialpad) [inline recipes/comboboxes/combobox_with_popover/combobox_with_popover.vue]: We shouldn't be focusing a non interactive element. [focusout](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusout_event) should still trigger when the input is focused as it captures anything inside of it as well this is the difference between [focusout](https://developer.mozilla.org/en-US/docs/Web/API/Element/focusout_event) and [blur](https://developer.mozilla.org/en-US/docs/Web/CSS/filter-function/blur)

**Review by braddialpad — CHANGES_REQUESTED** [review]

File: recipes/comboboxes/combobox_with_popover/combobox_with_popover.vue (hunk 2 above)
Comment (braddialpad) [reply recipes/comboboxes/combobox_with_popover/combobox_with_popover.vue]: I would suspect the problem is in this logic somewhere

```js
const isComboboxStillFocused = comboboxRefs.some((ref) => {
    return this.$refs[ref]?.contains(e.relatedTarget);
  }) || this.visuallyHiddenClose;
```

Comment (iropolo) [reply recipes/comboboxes/combobox_with_popover/combobox_with_popover.vue]: Final result after discussion

```
const isComboboxStillFocused = e.relatedTarget === null || comboboxRefs.some((ref) => {
        return this.$refs[ref]?.contains(e.relatedTarget);
      }) || this.visuallyHiddenClose;
```

**Review by braddialpad — APPROVED** [review]

**Pull request merged**